runners/spark/src/main/java/org/apache/beam/runners/spark/SparkRunner.java

```diff
@@ -216,6 +219,34 @@ public CompositeBehavior enterCompositeTransform(TransformTreeNode node) {
       return CompositeBehavior.ENTER_TRANSFORM;
     }
 
+    private boolean shouldDiffer(TransformTreeNode node) {
```

Comment:
And the only reason the SparkRunner has two translators is because for Spark 1.x streaming and batch have a separate API.